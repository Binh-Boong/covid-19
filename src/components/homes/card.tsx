import { ICountry } from "../../types/apis/country.module";

export const Card = ({ 
  country, 
}: {
  country: ICountry,
}):JSX.Element => {
  return (
    <div className="w-full p-3 text-gray-700 shadow">
      <h3 className="mb-3 font-bold text-center">{country.Country}</h3>
      <div className="flex justify-between w-full mt-1">
        <div className="text-blue-800">New Confirmed</div>
        <div className="text-blue-800">{country.NewConfirmed}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-blue-800">Total Confirmed</div>
        <div className="text-blue-800">{country.TotalConfirmed}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-red-700">New Deaths</div>
        <div className="text-red-700">{country.NewDeaths}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-red-700">TotalDeaths</div>
        <div className="text-red-700">{country.TotalDeaths}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-green-700">New Recovered</div>
        <div className="text-green-700">{country.NewRecovered}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-green-700">Total Recovered</div>
        <div className="text-green-700">{country.TotalRecovered}</div>
      </div>
    </div>
  );
}