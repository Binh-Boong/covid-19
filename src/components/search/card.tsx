import { ICountryStatus } from "../../types/apis/country-status.module";

export const Card = ({ 
  country, 
}: {
  country: ICountryStatus,
}):JSX.Element => {
  return (
    <div className="w-full p-3 text-gray-700 shadow">
      <h3 className="mb-3 font-bold text-center">{country.Date.split("T")[0]}</h3>
      <div className="flex justify-between w-full mt-1">
        <div className="text-blue-800">Confirmed</div>
        <div className="text-blue-800">{country.Confirmed}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-red-700">Deaths</div>
        <div className="text-red-700">{country.Deaths}</div>
      </div>
      <div className="flex justify-between w-full mt-1">
        <div className="text-green-700">Recovered</div>
        <div className="text-green-700">{country.Recovered}</div>
      </div>
    </div>
  );
}