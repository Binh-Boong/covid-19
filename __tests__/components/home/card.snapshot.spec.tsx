import renderer from "react-test-renderer";

import { Card } from "../../../src/components/homes/card";

const country = {
  Country: "VN",
  CountryCode: "VN",
  Slug: "vn",
  NewConfirmed: 1,
  TotalConfirmed: 2,
  NewDeaths: 3,
  TotalDeaths: 4,
  NewRecovered: 5,
  TotalRecovered: 6,
  Date: "29/09/2020",
  Premium: {
    CountryStats: {
      CountryISO: "CountryISO",
      Country: "Country",
      Continent: "Continent",
      Population: 1,
      PopulationDensity: 2,
      MedianAge: 3,
      Aged65Older: 4,
      Aged70Older: 5,
      ExtremePoverty: 6,
      GdpPerCapita: 7,
      CvdDeathRate: 8,
      DiabetesPrevalence: 9,
      HandwashingFacilities: 10,
      HospitalBedsPerThousand: 11,
      LifeExpectancy: 12,
      FemaleSmokers: 13,
      MaleSmokers: 14,
    },
  },
}

describe("Card Component", () => {
  it("renders correctly", () => {
      const card = renderer.create(
        <Card country={country} />
      ).toJSON();
    expect(card).toMatchSnapshot();
  });
});

export {}