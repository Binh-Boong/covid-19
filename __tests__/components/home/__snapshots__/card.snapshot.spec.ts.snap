// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Card Component renders correctly 1`] = `
<div
  className="w-full p-3 text-gray-700 shadow"
>
  <h3
    className="mb-3 font-bold text-center"
  >
    VN
  </h3>
  <div
    className="flex justify-between w-full mt-1"
  >
    <div
      className="text-blue-800"
    >
      New Confirmed
    </div>
    <div
      className="text-blue-800"
    >
      1
    </div>
  </div>
  <div
    className="flex justify-between w-full mt-1"
  >
    <div
      className="text-blue-800"
    >
      Total Confirmed
    </div>
    <div
      className="text-blue-800"
    >
      2
    </div>
  </div>
  <div
    className="flex justify-between w-full mt-1"
  >
    <div
      className="text-red-700"
    >
      New Deaths
    </div>
    <div
      className="text-red-700"
    >
      3
    </div>
  </div>
  <div
    className="flex justify-between w-full mt-1"
  >
    <div
      className="text-red-700"
    >
      TotalDeaths
    </div>
    <div
      className="text-red-700"
    >
      4
    </div>
  </div>
  <div
    className="flex justify-between w-full mt-1"
  >
    <div
      className="text-green-700"
    >
      New Recovered
    </div>
    <div
      className="text-green-700"
    >
      5
    </div>
  </div>
  <div
    className="flex justify-between w-full mt-1"
  >
    <div
      className="text-green-700"
    >
      Total Recovered
    </div>
    <div
      className="text-green-700"
    >
      6
    </div>
  </div>
</div>
`;
